src/shared/userSession/PageUserSession.ts

```diff
@@ -0,0 +1,148 @@
+import {
+    PageSettingsData,
+    PageSettingsUpdateRequest,
+} from 'shared/api/session-service/sessionServiceModels';
+import {
+    computed,
+    IReactionDisposer,
+    makeAutoObservable,
+    observable,
+    reaction,
+    toJS,
+} from 'mobx';
+import sessionServiceClient from 'shared/api/sessionServiceInstance';
+import { AppStore } from 'AppStore';
+import _ from 'lodash';
+import { IPageUserSession } from 'shared/userSession/IPageUserSession';
+import { PageSettingsIdentifier } from './PageSettingsIdentifier';
+
+export class PageUserSession<T extends PageSettingsData>
+    implements IPageUserSession<T> {
+    private _id: PageSettingsIdentifier;
+    private previousId: PageSettingsIdentifier | undefined;
+
+    /**
+     * User session as used in app, including user changes
+     */
+    private _userSettings: T | undefined;
+
+    /**
+     * User settings as stored in user session
+     */
+    private _savedUserSettings: T | undefined;
+
+    /**
+     * Saved user settings are loaded
+     */
+    private _isLoaded = false;
+
+    private reactionDisposers: IReactionDisposer[] = [];
+
+    constructor(
+        private appStore: AppStore,
+        private sessionServiceIsEnabled: boolean
+    ) {
+        makeAutoObservable(this);
+
+        this.reactionDisposers.push(
+            reaction(
+                () => [this.id, this.canSaveSession],
+                async () => {
+                    await this.fetchSessionUserSettings();
+                }
+            )
+        );
+    }
+
+    get isLoaded(): boolean {
+        return this._isLoaded;
+    }
+
+    public get id(): PageSettingsIdentifier {
+        return this._id;
+    }
+
+    @observable
+    public set id(id: PageSettingsIdentifier) {
+        this._id = id;
+    }
+
+    public get userSettings(): T | undefined {
+        return this._userSettings;
+    }
+
+    @observable
+    public set userSettings(userSettings: T | undefined) {
+        this._userSettings = userSettings;
+    }
+
+    @computed
+    public get canSaveSession() {
```

Comment:
userSessionEnabled

Reply:
Fixed in [9065f7e](https://github.com/cBioPortal/cbioportal-frontend/compare/240ae85eba446230bffd086012e309356e71f413..9065f7ea67dda257aad80c7a2b09cc88272492d9)